BIDS import/export, bet_premask bug fix, new modules

jooh opened the pull request:
BUG FIXES / IMPROVEMENTS:
* BIDS import takes new optional input arguments to specify tasks/sessions of interest (also useful for controlling the order in which data gets added to AA)
* BIDS import preserves the 'sub-' prefix, which is more consistent with the BIDS spec I think (so if you wanted to share the aa pipeline as a derivative the subject folder names would be sub-x rather than just numbers).
* BIDS export can now handle subject names with or without sub- prefix
* fix bug in bet_premask (Matlab is not very good at testing the existence of relative paths, so you have to check for abs path first).

NEW MODULES:
* aamod_meanstructural - does what it says on the tin.
* aamod_firstlevel_model_1/2/3 - split up the first level modeling so that custom processing can be added before the model is estimated. The standard aamod_firstlevel_model remains available and is sufficient for most people I imagine.
* aamod_firstlevel_model_subrunsplit - an example of what you can do when the first level model is split into stages. Here we accommodate 'sub run' splits - cases where independent runs of the experiment were acquired without stopping the scanner. This module makes it possible to split the SPM model up so that these sub runs are modelled as independent sessions. This is a useful initial stage e.g. for MVPA.

TODO:
* tests for the above! I have run all of the above through my own code and it all works fine. Once my current dataset is posted on openfmri that can perhaps be added as another BIDS test AA script that would include the above modules.

tiborauer commented:
Could you, please, make sure that the function header in aamod_meanstructural.m (line 1) is consitent with the function name?

jooh commented:
Done.

tiborauer commented:
I have tested the relevant functions (reading and writing BIDS), and they are working as usual. Though, I have not tested the new features and modules. I suggest to come up with relevant examples. However, if you are confident enough, I am happy to merge the PR.

jooh commented:
I have tested so I think we should be ok.

On 22 June 2017 at 10:35, Tibor Auer <notifications@github.com> wrote:

> I have tested the relevant functions (reading and writing BIDS), and they
> are working as usual. Though, I have not tested the new features and
> modules. I suggest to come up with relevant examples. However, if you are
> confident enough, I am happy to merge the PR.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rhodricusack/automaticanalysis/pull/115#issuecomment-310329238>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABfNs6EGy2dDJI3PcPatQ-kB4hzQIVsmks5sGjV4gaJpZM4OAzDw>
> .
>



-- 
Johan Carlin, PhD
Research Associate
MRC Cognition and Brain Sciences Unit
Cambridge, UK

jones-michael-s commented:
Small glitch: I believe there's a typo in the report generation in the firstlevel_model modules -- the jpegs that get created during analysis are named diagnostics_... (note the "s") and the report generation looks for files named diagnostic_... The (current) single-file version of aamod_firstlevel_model.m has the same glitch.

There's also currently quite a number of firstlevel module variants in the distribution. As a new user, I found this rather confusing. Might we want to think about organizing these (or culling ones that are more "experimental") in the spirit of trying to make aa more user friendly?

jooh commented:
Ok, I had no idea reports work in that way. I ran a grep diagnostics_ aa_modules/*.m and changed accordingly. Better now?

Re models, I think fMRI necessarily involves a huge number of custom first level models. That's where almost all the theoretically interesting stuff happens (the second level model is usually just RFX on the summary stats). The alternative is to try to cover all the unusual model cases in one monolithic aamod_firstlevel_model, but I think that's actually more confusing. For one thing you would need to look at the module settings to know what kind of model was run, whereas with custom modules you can see immediately from the task list what was done.

So I think we should have many first level models, but perhaps we need to add more documentation to clarify exactly what each variant is doing. For instance, the models I added now have lots of comments discussing what they are for, and pointing out that you may prefer aamod_firstlevel_model for standard use cases.

jones-michael-s commented:
> Ok, I had no idea reports work in that way. I ran a grep diagnostics_ aa_modules/*.m and changed accordingly. Better now?

I checked the new diffs and they look good to me. Thanks.

Also, I see your point about providing useful custom modules (for first level analysis and in general).

jooh commented:
So we are all good to merge then @tiborauer?
